staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/stdlib.go

```diff
@@ -0,0 +1,585 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"net/url"
+	"regexp"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/common/types/traits"
+	"github.com/google/cel-go/interpreter/functions"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+
+	"k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model"
+)
+
+// Stdlib provides a CEL function library extension that supplements the CEL standard function library.
+//
+// isSorted
+//
+// Returns true if the provided list of comparable elements is sorted, else returns false.
+//
+//     <list<T>>.isSorted() <bool>, T must be a comparable type
+//
+// Examples:
+//
+//     [1, 2, 3].isSorted()  // return true
+//
+//
+// sum
+//
+// Returns the sum of the elements of the provided list. Supports CEL number (int, uint, double) and duration types.
+//
+//     <list<T>>.sum() <T>, T must be a numeric type or a duration
+//
+// Examples:
+//
+//     [1, 3].sum() // returns 4
+//     [].sum() // returns 0
+//
+//
+// min / max
+//
+// Returns the minimum/maximum valued element of the provided list. Supports all comparable types.
+// If the list is empty, an error is returned.
+//
+//     <list<T>>.min() <T>, T must be a comparable type
+//     <list<T>>.max() <T>, T must be a comparable type
+//
+// Examples:
+//
+//     [1, 3].min() // returns 1
+//     [1, 3].max() // returns 3
+//     [].min() // error
+//
+//
+// indexOf / lastIndexOf
+//
+// Returns the positional index of the provided element in the list. Supports all equatable types.
+// If the element is not found, -1 is returned.
+//
+//     <list<T>>.indexOf(<T>) <int>
+//     <list<T>>.lastIndexOf(<T>) <int>
+//
+// Examples:
+//
+//     [1, 2, 2, 3].indexOf(2) // returns 1
+//     [1, 2, 2, 3].lastIndexOf(2) // returns 2
+//     [1].indexOf(2) // returns -1
+//
+//
+// find / findAll
+//
+// Returns substrings that match the provided regular expression. find returns the first match. findAll may optionally
+// be provided a limit. If the limit is set and >= 0, no more than the limit number of matches are returned.
+//
+//     <string>.find(<string>) <string>
+//     <string>.findAll(<string>) <list <string>>
+//     <string>.findAll(<string>, <int>) <list <string>>
+//
+// Examples:
+//
+//     "abc 123".find('[0-9]*') // returns '123'
+//     "abc 123".find('xyz') // returns ''
+//     "123 abc 456".findAll('[0-9]*') // returns ['123', '456']
+//     "123 abc 456".findAll('[0-9]*', 1) // returns ['123']
+//     "123 abc 456".findAll('xyz') // returns []
+//
+//
+// url
+//
+// Converts a string to a URL or results in an error if the string is not a valid URL.
+//
+//     url(<string>) <URL>
+//
+//
+// isURL
+//
+// Returns true if a string is a valid URL.
+//
+//     <string>.isURL() <bool>
```

Comment:
is there a reason this isn't `isURL(<string>)`? are there other `is___()` functions added to string already we're mimicking

Reply:
I can get behind `isURL(<string>)`. There isn't any precedence that I'm aware of. I've updated the code to match.

Thinking about it. I feel like if we're going to support the string->URL test we should have string-><everything a string can convert to> test functions to match...

@TristonianJones is there a way to test if a `string->double` conversion is safe before attempting it?

Reply:
discussed offline and agreed the use cases for things like `isDouble` are not particularly compelling, I think this is okay as is.

Reply:
@jpbetz Sorry, are you asking if the conversion will succeed because the string is well-formatted? If so, there's no specific test or guarantee for this in CEL